### PR TITLE
sci-libs/cantera: 2.4.0-r2 revision bump

### PR DESCRIPTION
**Pull request by band-a-prend**
Backported from upstream:
* Fix python installation path using `libdirname` env variable 
instead of auto detection of python installation path
* Fix build against `>=sci-libs/sundials-4.0`

@mgorny 
please review this pull request

### CODE REVIEW

**Comment by band-a-prend** [comment]
@juippis 
could you review this pull request?

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+	<sci-libs/sundials-5.0.0:0=
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
```
Comment (juippis) [inline sci-libs/cantera/cantera-2.4.0-r2.ebuild]: If this get's linked during build, it should have a subslot binder.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	${PYTHON_DEPS}
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+	<sci-libs/sundials-5.0.0:0=
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
```
Comment (juippis) [inline sci-libs/cantera/cantera-2.4.0-r2.ebuild]: This is slotted.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	dev-libs/boost
+	dev-libs/libfmt
```
Comment (juippis) [inline sci-libs/cantera/cantera-2.4.0-r2.ebuild]: This has a subslot, does it get linked during build?

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+FORTRAN_NEEDED=fortran
+FORTRAN_STANDARD=90
+
+inherit desktop fortran-2 python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
```
[the remark this answers is not in the document]
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: This [page](https://cantera.org/community.html#about-the-cantera-community) states:

>  ... Cantera is licensed under a permissive 3-Clause BSD license ...

Also the linked license.txt is similar to that BSD license.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 2 above)
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: Currently there is only one slot of eigen in the portage tree so I don't know will the API of used functions be changed in future slots. Slot 2 of eigen was droped before cantera package appeared in portage tree.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+FORTRAN_NEEDED=fortran
+FORTRAN_STANDARD=90
+
+inherit desktop fortran-2 python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
```
[the remark this answers is not in the document]
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: Fixed in new commit.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 1 above)
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: The [dependencies page](https://cantera.org/compiling/dependencies.html#other-required-software) mentions that:
> Only the “header-only” portions of Boost are required. Cantera does not currently depend on any of the compiled Boost libraries. 

So Boost is not linked. All tests passed for cantera compiled against boost-1.70.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 3 above)
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: libfmt isn't linked too, only headers used. Tests are also passed with libfmt-6.0

**Comment by band-a-prend** [comment]
@juippis 
please don't merge this PR, I want to check some things again

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 2 above)
Comment (juippis) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: To be sure it doesn't and to help portage, you should add it though.

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 4 above)
[the remark this answers is not in the document]
Comment (juippis) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: Sure, just wanted you to confirm :)

**Comment by juippis** [comment]
Could you rebase against fresh master so we get a green light from the CI bot?

File: sci-libs/cantera/cantera-2.4.0-r2.ebuild (hunk 2 above)
Comment (band-a-prend) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: Do you mean the addition of restriction by specific SLOT (dev-cpp/eigen:3) here?

Comment (juippis) [reply sci-libs/cantera/cantera-2.4.0-r2.ebuild]: or `:*` if you **know** it won't break.

**Comment by band-a-prend** [comment]
@juippis 
there are too many broken packages again in the CI report but the `sci-libs/cantera` is absent among them

**Comment by juippis** [comment]
But you need to fetch latest master, then rebase your branch against that master. I want to see it doesn't break the current tree for sure.

**Comment by band-a-prend** [comment]
@juippis 
Done. Thank you for hint about rebase.

**Pull request closed without merging**